### DISCUSSION TITLE
Possible bug in ezdxf open_uniform_knot_vector function

### DISCUSSION THREAD

**Discussion by subsurfaceiodev** [mozman/ezdxf#1028] [Issues]
Hi @mozman ! 
[ezdxf_spline_issue.zip](https://github.com/mozman/ezdxf/files/14227007/ezdxf_spline_issue.zip) contains the following files:

1. **to_import.dxf** created in AutoCAD containing 3 splines: 2 individual splines on the left and the product of joining those splines as a single spline on the right
![image](https://github.com/mozman/ezdxf/assets/109676812/acf70ed5-d4b4-4308-8a5d-080a1a9d0a6f)

2. **spline_mwe.py** containing the following code:
```python
import ezdxf
from ezdxf import zoom
from ezdxf.entities import Spline

doc = ezdxf.readfile('to_import.dxf')
msp = doc.modelspace()

doc_new = ezdxf.new()
msp_new = doc_new.modelspace()

for e in msp:
    if isinstance(e, Spline):
        spline = msp_new.add_open_spline(
            control_points=e.control_points,
            # knots=e.knots,
            degree=2,
        )
        # actual knots
        knots = e.knots

        # open_uniform_knot_vector
        knots_ezdxf = spline.knots
        print(f'{knots=}, {knots_ezdxf=}')
zoom.extents(msp_new)
doc_new.saveas('exported.dxf')
```
3. **exported.dxf** which is the product of the above code: 
![image](https://github.com/mozman/ezdxf/assets/109676812/ccfc80e3-2401-47d5-ae15-9e66641ac2a8)

As you may see, the right spline is not exported correctly when commenting line 15 of the above code (`knots=e.knots`), we noticed when knots are not provided `add_open_spline` function calls `open_uniform_knot_vector` to calculate them. Said calculation works fine for the left splines, but for the right spline you may see AutoCAD knots are: `[0.0, 0.0, 0.0, 1.0, 1.0, 2.0, 2.0, 2.0]` while calculated knots by ezdxf are `[0.0, 0.0, 0.0, 1.0, 2.0, 3.0, 3.0, 3.0]` which seems to be a bug causing exported right spline to not look correct.

**Comment by mozman** [comment]
This is a special spline configuration and not an open B-spline with uniform knot distribution because the interior knots (1.0) are repeated and the curve passes the third control vertex. See [definition](https://www.cl.cam.ac.uk/teaching/1999/AGraphHCI/SMAG/node4.html) for open uniform knots.

![image](https://github.com/mozman/ezdxf/assets/4230966/c132664d-2112-45a4-8f3c-f2a4621635a4)

This means that this spline configuration cannot be created using the`add_open_spline()` factory method. To get the same spline you have to use the original knots as done by commenting out the line `knots=e.knots`.

**Reply by subsurfaceiodev** [reply]
It makes sense, is there any function in ezdxf to get said special spline configuration knots given control points?
